app/main.py

```diff
@@ -52,17 +52,23 @@ async def main():
             for table in storage.tables.values():
                 if table._meta.tablename == config.table_name:
                     # visible columns
-                    visible_columns = [
-                        column
-                        for column in table._meta.columns
-                        if column._meta.name in config.visible_columns
-                    ]
+                    try:
+                        visible_columns = [
+                            column
+                            for column in table._meta.columns
+                            if column._meta.name in config.visible_columns
+                        ]
+                    except TypeError:
+                        visible_columns = table._meta.columns
```

Comment:
Here you can use `table._meta.get_column_by_name` instead.

If an exception is raised, I think we should tell the user that the column isn't recognised, rather than defaulting to all columns.

Reply:
I'm sorry, but I don't understand. This only prevents a `TypeError` if the user only wants the `table_name` in the `yaml` file as requested in one of the previous comments.

> So if the user just wants to show a table, and isn't bothered about passing in any config options, we just show the tables with a default config.

If the user entered the column name incorrectly (only one column) like this 
```yaml
- table_name: actor
    visible_columns:
      - first_na # wrong column name
```
we return the default config (all columns). 

If the user correctly entered multiple columns and one or more columns is written incorrectly like this
```yaml
- table_name: actor
    visible_columns:
      - first_na # wrong
      - last_name # correct
```
we return only the correctly written columns (in this case `last_name`) and the incorrectly written column is omitted.

Reply:
Rather than letting the code raise a `TypeError` you can just check `config.visible_columns is not None`. It was confusing to me why we were trying to catch a `TypeError`.

I suggested using `table._meta.get_column_by_name` to check if the column exists, because it raises an exception if a match isn't found.

```python
if config.visible_columns:
    visible_columns = [
        table._meta.get_column_by_name(column_name) for column_name in config.visible_columns
    ]
else:
    visible_columns = table._meta.columns
```

I think if any of the column names are misspelt, we should just raise an error. Otherwise the user won't know they have a typo.

Reply:
Yeah. You are right. I didn't think about it. Thanks for the suggestion. That's much better. I'll change that tomorrow.